File: src/jvmMain/java/core/Sleep.java
package core;

/* Simone 11/05/13 9.59 */

import org.joda.time.Duration;

public class Sleep  {
    public static Sleep instance = new Sleep();

    public void sleep(long millis) {
        sleepStatic(millis);
    }

    public static void sleepStatic(long millis) {
        try {
            Thread.sleep(millis);
        } catch (InterruptedException ex) {
            throw new RuntimeException(ex);
        }
    }

    public static void sleepStatic(Duration duration)
    {
        sleepStatic(duration.getMillis());
    }

    public static void forever() {
        while (true)
            sleepStatic(10000);
    }
}

File: satd-classifier/src/jvmMain/kotlin/core/Sleep.kt
package core

import org.joda.time.Duration

/* Simone 11/05/13 9.59 */   class Sleep {
    fun sleep(millis: Long) {
        sleepStatic(millis)
    }

    companion object {
        var instance = Sleep()
        fun sleepStatic(millis: Long) {
            try {
                Thread.sleep(millis)
            } catch (ex: InterruptedException) {
                throw RuntimeException(ex)
            }
        }

        fun sleepStatic(duration: Duration) {
            sleepStatic(duration.millis)
        }

        fun forever() {
            while (true) sleepStatic(10000)
        }
    }
}